assets/src/edit-story/components/inspector/prepublish/prepublishCheckpointState.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const PPC_CHECKPOINT_STATE = {
+  UNAVAILABLE: 'unavailable',
+  ONLY_RECOMMENDED: 'recommended',
+  ALL: 'all',
+};
+
+export const PPC_CHECKPOINT_ACTION = {
+  ON_DIRTY_STORY: 'story is no longer empty',
+  ON_PUBLISH_CLICKED: 'publish button is pressed',
+  ON_STORY_HAS_5_PAGES: 'story has minimal suggested amount of pages',
+};
+
+const machine = {
+  [PPC_CHECKPOINT_STATE.UNAVAILABLE]: {
+    [PPC_CHECKPOINT_ACTION.ON_DIRTY_STORY]:
+      PPC_CHECKPOINT_STATE.ONLY_RECOMMENDED,
+    [PPC_CHECKPOINT_ACTION.ON_PUBLISH_CLICKED]: PPC_CHECKPOINT_STATE.ALL,
+  },
+  [PPC_CHECKPOINT_STATE.ONLY_RECOMMENDED]: {
+    [PPC_CHECKPOINT_ACTION.ON_STORY_HAS_5_PAGES]: PPC_CHECKPOINT_STATE.ALL,
+    [PPC_CHECKPOINT_ACTION.ON_PUBLISH_CLICKED]: PPC_CHECKPOINT_STATE.ALL,
+  },
```

Comment:
I'm fine with leaving the actions reflective of the events, but since we'll have an event system, we could just make this one action and make the action describe the transition vs the triggering event:
```
  [PPC_CHECKPOINT_STATE.ONLY_RECOMMENDED]: {
    [PPC_CHECKPOINT_ACTION.EVALUATE_ALL]: PPC_CHECKPOINT_STATE.ALL,
  },
```
Then from the story events we want to have:
```
    useStoryTriggerListener(
      STORY_EVENTS.onPageChange, // This can be called whatever
      useCallback((story) => {
        if (story.pages.length >= 5) {
          dispatch(PPC_CHECKPOINT_ACTION.EVALUATE_ALL);
        }
      }, [])
    );
    
    useStoryTriggerListener(
      STORY_EVENTS.onPublish,
      useCallback(() => {
        dispatch(PPC_CHECKPOINT_ACTION.EVALUATE_ALL);
      }, [])
    )
```

Kinda one half dozen of the other tho 🤷

Reply:
Totally a 6, half dozen - the only reason why I gravitate towards having event based is because it's a really straight forward object that is super easy to figure out what is getting called with x happens. Right now both feel clean and simple enough and we shouldn't design for a crazy possible future, but I'm curious as this grows and more checks are added possibly what version keeps things a little cleaner.

Reply:
Awesome. yea good point with transparency with actions. Cool, yea I can't imagine it growing much beyond what we got (separating the types of messages in ppc), but def good to keep an eye on as we go. But cool! let's leave it as is for now